File: SophomoreYear/CMPS12M/Lab3/fileReverse.c

/*
 * ------------------
 * Bryan Lopez
 * blopez24 (1612626)
 * ------------------
 * fileReverse.c
 * Takes two command line arguments naming the input and output file.
 * Reads each word from the input file, then prints it backwards on a line
 * by itself.
 */

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

/* Reverses the string argument */
void stringReverse(char* s){
	int i;					/* Starting index */
	int j = strlen(s) - 1;	/* Length of s - 1 */
	char temp;				/* Temporary holds a letter */
	
	for(i = 0; i < j; i++){
		temp = s[i];		/* Holds letter at i */
		s[i] = s[j];		/* Swaps letter at i with j */
		s[j] = temp;		/* String at j is assign to temp */
		j--;
	}
}

int main(int argc, char* argv[]){
	FILE* in;		/* file handle for input */
	FILE* out;		/* file handle for output */
	char word[256];	/* char array to store words from input file */
	
	/* check command line for correct number of arguments */
	if( argc != 3){
		printf("Usage: %s <input file> <output file>\n", argv[0]);
		exit(EXIT_FAILURE);
	}
	
	/* open input file for reading */
	in = fopen(argv[1], "r");
	if( in == NULL ){
		printf("Unable to read from file %s\n", argv[1]);
		exit(EXIT_FAILURE);
	}
	
	/* open output file for writing */
	out = fopen(argv[2], "w");
	if( out==NULL ){
		printf("Unable to write to file %s\n", argv[2]);
		exit(EXIT_FAILURE);
	}
	
	/* read words from input file, uses stringReverse(), and prints on separate lines to output file */
	while( fscanf(in, " %s", word) != EOF ){
		stringReverse(word);
		fprintf(out, "%s\n", word);
	}

	/* close input and output files */
	fclose(in);
	fclose(out);
	
	return(EXIT_SUCCESS);
}

File: SophomoreYear/CMPS12M/Lab2/FileReverse.java
//--------------------------------------------------------------------------
// Bryan Lopez
// blopez24 (1612626)
//--------------------------------------------------------------------------
// FileReverse.java
// Takes two command line arguments. Reads each line of input, parse the tokens, then prints each token backwards to the output file on a line by itself.
//--------------------------------------------------------------------------

import java.io.*;
import java.util.Scanner;

class FileReverse 
{
	public static void main(String[] args) throws IOException
	{
		//  Check number of command line arguments is at least 2
		if(args.length < 2)
		{
			System.out.println("Usage: java –jar FileReverse.jar <input file> <output file>");
			System.exit(1);
			}

		// Open files
		Scanner in = new Scanner(new File(args[0]));
		PrintWriter out = new PrintWriter(new FileWriter(args[1]));
		
		// Read lines from in, extract and print tokens from each line
		while( in.hasNextLine() )
		{
			// Trim leading and trailing spaces, then add one trailing space so split works on blank lines
			String line = in.nextLine().trim() + " "; 
			
			// Split line around white space 
			String[] token = line.split("\\s+");  

			// Print out tokens       
			int n = token.length;
              
			for(int i=0; i<n; i++)
			{
				out.println(stringReverse(token[i]));
			}
		}

            // Close files
            in.close();
            out.close();
         }

	// Returns a string that is the reversal of s
	public static String stringReverse(String s)
	{
		// Checks if s is at least 1 
		if(s.length() <= 1)
			return s;
		// Returns last letter and calls the method again without the last letter
		return s.charAt(s.length()-1) + stringReverse(s.substring(0, s.length()-1));
	}	
}
